### ISSUE TITLE
A disabled BlazorNavLink still activates the router when clicked.

### ISSUE THREAD

**Issue by MarkStega** [chanan/BlazorStrap#13]
```
                <NavItem>
                    <BlazorNavLink href="logoffview" IsDisabled="true">Log off</BlazorNavLink>
                </NavItem>
```
This link is rendered in the disabled color, however when I select it I am routed to 'logoffview'.
I can see that the IsDisabled is part of the rendered code, but it isn't functioning as I would expect.
```
                        builder4.CloseComponent();
                        builder4.AddContent(112, "\n                ");
                        builder4.OpenComponent<BlazorStrap.NavItem>(113);
                        builder4.AddAttribute(114, "ChildContent", (Microsoft.AspNetCore.Blazor.RenderFragment)((builder5) => {
                            builder5.AddContent(115, "\n                    ");
                            builder5.OpenComponent<BlazorStrap.BlazorNavLink>(116);
                            builder5.AddAttribute(117, "href", "logonview");
                            builder5.AddAttribute(118, "IsDisabled", Microsoft.AspNetCore.Blazor.Components.RuntimeHelpers.TypeCheck<System.Boolean>(true));
                            builder5.AddAttribute(119, "ChildContent", (Microsoft.AspNetCore.Blazor.RenderFragment)((builder6) => {
                                builder6.AddContent(120, "Logon");
                            }
                            ));
                            builder5.CloseComponent();
                            builder5.AddContent(121, "\n                ");
                        }
                        ));
                        builder4.CloseComponent();
```

**Comment by chanan** [comment]
I am not sure, but it seems to me that the bootstrap site might be "cheating" :)

Looking at: https://getbootstrap.com/docs/4.1/components/navs/ none of the links actually change the page and our markup is pretty much the same. Maybe they have a javascript on all links to disable them? So, I think what you get from disabled is just the style, but its up to you to disable the onclick event.

But, I might be wrong, thoughts on this?

**Comment by LorenDorez** [comment]
@chanan Yes i found this one out myself when i started using BS4. on the demos they run a script to disable all empty links which makes the real world example of it not true. Maybe just remove the Href on Disable links? or use a JS routine to disable them. Either way i would atleast update the documentation to state this. Bootstrap is misleading as they only state it add a disabled _STYLE_ to the element.

Reference
https://github.com/twbs/bootstrap/blob/8fa0d3010112dca5dd6dd501173415856001ba8b/site/docs/4.3/assets/js/src/application.js#L40

**Comment by chanan** [comment]
@LorenDorez Removing the href when disabled is true should be easy enough. I will try to get to that tonight.

**Comment by LorenDorez** [comment]
@chanan thanks. No rush was just a suggestions so people dont have to wire up JS. 

Just though of an issue with my suggestion. If they want to re-enable the link the href is now gone so maybe store in a temp. parameter, so it can be re-enabled? again just some ideas

**Comment by LorenDorez** [comment]
@chanan No rush. i was just tying to help with some ideas  for solutions :) So since BS4 doesnt really handle this properly its not really a bug more of a Feature-Enhancement

**Comment by peterblazejewicz** [comment]
The BS 4 ships with rule that disables interaction when `.disabled` class has been added. So this should trap the events.
The specs are lacking inert state for interactive `a` tag used in nav bar links:
https://github.com/whatwg/html/pull/4288

**Comment by chanan** [comment]
Fixed in 1.0.0-preview6-21
